src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,137 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    createClients.call(this);
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
+   * @returns {Promise}
+   */
+  callWithRequest = (req = {}, endpoint, clientParams = {}, options = {}) => {
```

Comment:
@tylersmalley But think about when this is getting executed... it's at class definition time, so there is no "this".  The current setup is behaving like the follow:

```js
function Cluster() {
  this.callWithRequest = () => {
    // this is an instance of Cluster
  };
}
```

even though the syntax should essentially be mapped to:

```js
const callWithRequest = () => {
  // this is whatever the current module context is
};
function Cluster() {
  this.callWithRequest = callWithRequest;
}
```

The only thing I can think of is that the instance property transpiler is actually putting the arrow function inside the constructor before transpiling the arrow function to a `.bind(this)` (or equivalent), but that doesn't seem *right* to me.  It would surprise me if that's what the property spec defines as the correct evaluation behavior, so I'm worried that if we go down this route, someday we'll pull this es6 transpiling rule and suddenly the class won't work anymore.